metals/src/main/scala/scala/meta/internal/metals/SyntheticDecorationsProvider.scala

```diff
@@ -0,0 +1,200 @@
+package scala.meta.internal.metals
+
+import scala.annotation.nowarn
+import scala.collection.mutable
+import scala.{meta => m}
+
+import scala.meta.internal.decorations._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.pc.DecorationKind
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.SyntheticDecoration
+import scala.meta.tokens.{Token => T}
+
+import org.eclipse.{lsp4j => l}
+
+final class SyntheticDecorationsProvider(
+    params: m.pc.VirtualFileParams,
+    trees: Trees,
+    userConfig: () => UserConfiguration,
+) {
+
+  val path: AbsolutePath = params.uri().toAbsolutePath
+
+  lazy val (withoutTypes, methodPositions) = declarationsWithoutTypes()
```

Comment:
is methodPositions used?

Reply:
Yes, for methods we see if it lacks type annotation based on `withoutTypes`, but we insert the decoration in `methodPosition`. Maybe I will rename it to `methodTpeAnnotPosition` if it's clearer

Reply:
Ach ok, I missed it